src/test/java/com/scottlogic/filters/DatesFilterTest.java

```diff
@@ -0,0 +1,159 @@
+        OffsetDateTime maxDate = OffsetDateTime.of(2020, 12, 31, 7, 12, 3, 0, ZoneOffset.UTC);
+
+        UserPost userPost1 = createUserPost(OffsetDateTime.of(20, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost2 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost3 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 1, ZoneOffset.UTC));
+        UserPost userPost4 = createUserPost(OffsetDateTime.of(2012, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost containsMaxDate = createUserPost(maxDate);
+
+
+        DatesFilter filterByDates = new DatesFilter(OffsetDateTime.of(2020, 1, 1, 7, 12, 3, 0, ZoneOffset.UTC),
+                maxDate);
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, userPost3, userPost4, containsMaxDate);
+        List<UserPost> actualResult = filterByDates.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(userPost2, userPost3, containsMaxDate);
+        Assert.assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void filter_borderDatesMin_borderDatesIncludedReturnedSecondsIgnored() {
+        OffsetDateTime minDate = OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC);
+
+        UserPost userPost1 = createUserPost(OffsetDateTime.of(20, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost minDatePlusSeconds = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 4, 0, ZoneOffset.UTC));
+        UserPost hasMinDate = createUserPost(minDate);
+        UserPost userPost4 = createUserPost(OffsetDateTime.of(2012, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost5 = createUserPost(OffsetDateTime.of(2020, 12, 31, 7, 12, 3, 0, ZoneOffset.UTC));
+
+        DatesFilter filterByDates = new DatesFilter(minDate, OffsetDateTime.of(2020, 12, 31, 7, 12, 3, 0, ZoneOffset.UTC));
+        List<UserPost> userPosts = Arrays.asList(userPost1, minDatePlusSeconds, hasMinDate, userPost4, userPost5);
+        List<UserPost> actualResult = filterByDates.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(minDatePlusSeconds, hasMinDate, userPost5);
+        Assert.assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void filter_validInput_inputListNotMutated() {
+        DatesFilter filterByDates = new DatesFilter(OffsetDateTime.of(2020, 1, 1, 7, 12, 3, 0, ZoneOffset.UTC),
+                OffsetDateTime.of(2020, 12, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+
+        UserPost userPost1 = createUserPost(OffsetDateTime.of(20, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+        UserPost userPost2 = createUserPost(OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC));
+
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2);
+        List<UserPost> userPostsCopy = Arrays.asList(userPost1, userPost2);
+        filterByDates.filter(userPosts);
+        Assert.assertEquals(userPostsCopy, userPosts);
+    }
```

Comment:
I think we're missing a test where there's a null date in one of the posts.

Reply:
I'm not able to test that because null date is not allowed by the Object